fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/RecordQueryPlanner.java

```diff
@@ -517,9 +566,38 @@ private ScoredPlan planIndex(@Nonnull PlanContext planContext, @Nonnull QueryCom
                 }
             }
         }
+
         return p;
     }
 
+    private ScoredPlan computeIndexFilters(@Nonnull PlanContext planContext, @Nonnull final ScoredPlan plan) {
+        if (plan.plan instanceof RecordQueryPlanWithIndex) {
```

Comment:
A `RecordQueryScanPlan` could treat all filters as index filters and not residuals, I think.